### PR TITLE
Add interval(x0) in shortcut constructor

### LINKED ISSUE #21: Shortcut constructor for independent variable
```julia
julia> using TaylorModels
[ Info: Recompiling stale cache file /Users/forets/.julia/compiled/v1.1/TaylorModels/yPZW1.ji for TaylorModels [a2574a24-d24b-5d8e-9d8b-7eb85f89fcf9]

julia> TaylorModel1(2, Interval(0.5, 0.5), Interval(0.0, 1.0))
 [0.5, 0.5] + [1, 1] t + [0, 0]

julia> TaylorModel1(2, 0.5, Interval(0.0, 1.0))
ERROR: MethodError: no method matching TaylorModel1(::Taylor1{Interval{Float64}}, ::Interval{Float64}, ::Float64, ::Interval{Float64})
Closest candidates are:
  TaylorModel1(::Taylor1{T}, ::Interval{S}, ::Interval{S}, ::Interval{S}) where {T, S} at /Users/forets/.julia/dev/TaylorModels/src/constructors.jl:33
  TaylorModel1(::Integer, ::Any, ::Interval{T}) where T at /Users/forets/.julia/dev/TaylorModels/src/constructors.jl:36
  TaylorModel1(::Interval{T}, ::Integer, ::Interval{T}, ::Interval{T}) where T at /Users/forets/.julia/dev/TaylorModels/src/constructors.jl:41
  ...
Stacktrace:
 [1] TaylorModel1(::Int64, ::Float64, ::Interval{Float64}) at /Users/forets/.julia/dev/TaylorModels/src/constructors.jl:36
 [2] top-level scope at none:0
```

Changing the penultimate argument in [this line](https://github.com/JuliaIntervals/TaylorModels.jl/blob/master/src/constructors.jl#L37), `x0`, to `interval(x0)` one can use the short-cut constructor `TaylorModel1(2, 0.5, Interval(0.0, 1.0))`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #21.

### CODE REVIEW

**Comment by mforets** [comment]
Hmm does it fail in windows 32bit because of the `0.0` in `@test tv == TaylorModel1(5, 0.0, ii0)`? I don't know if it makes a difference to use `@test tv == TaylorModel1(5, Float64(0.0), ii0)`.

File: test/TM1.jl [hunk 1]
```diff
@@ -45,6 +45,7 @@ end
         tv = TaylorModel1{Interval{Float64},Float64}(Taylor1(Interval{Float64},5), x0, x0, ii0)
         @test tv == TaylorModel1(Taylor1(Interval{Float64},5), x0, x0, ii0)
         @test tv == TaylorModel1(5, x0, ii0)
+        @test tv == TaylorModel1(5, 0.0, ii0)
```
Comment (mforets) [inline test/TM1.jl]:
```suggestion
        @test tv == TaylorModel1(5, Float64(0.0), ii0)
```

**Comment by mforets** [comment]
> I don't know if it makes a difference to use @test tv == TaylorModel1(5, Float64(0.0), ii0), but i could try.

No, it does not help with this problem. I dunno.

**Comment by lbenet** [comment]
What ever happens in widows x86 is beyond my understanding; a *simple* solution to get tests passing is to decide not to care about x86, and only care about x64.

**Comment by dpsanders** [comment]
Note that you can (and should) use 
```
git rebase master
```
to update a branch on top of current `master`.

**Comment by dpsanders** [comment]
Then you will need to force push the branch (`git push -f`).

**Comment by mforets** [comment]
Done, there is now only 1 commit in the history of this PR.

**Comment by lbenet** [comment]
Tests are not passing; do you understand why? I guess this may be related that your initial PR originated in a (now outdted) master. Perhaps, rebasing to current master is the way to go.

File: src/constructors.jl [hunk 2]
```diff
@@ -34,7 +34,7 @@ for TM in tupleTMs
 
         # Short-cut for independent variable
         $(TM)(ord::Integer, x0, dom::Interval{T}) where {T} =
-            $(TM)(interval(x0) + Taylor1(Interval{T}, ord), zero(dom), x0, dom)
+            $(TM)(interval(x0) + Taylor1(Interval{T}, ord), zero(I), interval(x0), dom)
```
Comment (mforets) [inline src/constructors.jl]:
```suggestion
            $(TM)(interval(x0) + Taylor1(Interval{T}, ord), zero(dom), interval(x0), dom)
```

File: src/constructors.jl [hunk 3]
```diff
@@ -34,7 +34,7 @@ for TM in tupleTMs
 
         # Short-cut for independent variable
         $(TM)(ord::Integer, x0, dom::Interval{T}) where {T} =
-            $(TM)(interval(x0) + Taylor1(Interval{T}, ord), zero(dom), x0, dom)
+            $(TM)(interval(x0) + Taylor1(Interval{T}, ord), zero(dom), interval(x0), dom)
```
Comment (lbenet) [inline src/constructors.jl]: It just occurred to me that we could make this method a bit more general, e.g., to return a TaylorModel with `Float64` polynomial coefficients. Then, this method would need to be:
```julia
$(TM)(ord::Integer, x0::T, dom::Interval{T}) where {T} =
    $(TM)(x0 + Taylor1(T, ord), zero(dom), interval(x0), dom)
```
Let me know if you want to do this, or if I simply go ahead and merge (and latter we do the change).

**Comment by lbenet** [comment]
I'll merge this, and if a change is needed, we can do it in another PR.

Thanks a lot @mforets !

**Comment by mforets** [comment]
> It just occurred to me that we could make this method a bit more general, e.g., to return a TaylorModel with Float64 polynomial coefficients.

OK, i see. What do you think about allowing both versions? Like:

```julia
julia> TaylorModel1(2, 5.0, Interval(4.0, 6.0))
 5.0 + 1.0 t + [0, 0]

julia> TaylorModel1(2, Interval(5.0), Interval(4.0, 6.0)) 
[5, 5] + [1, 1] t + [0, 0]
```

**Comment by lbenet** [comment]
Yes, but I am not sure if this extends to several variables. In any case, this is merged, and if posible, we should have this in a new PR. Thanks a lot for the contribution!

**Pull request merged**